core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -3034,7 +3034,8 @@ private void setBuildsAndWorkspacesDir() throws IOException, InvalidBuildsDir {
 
         String newWorkspacesDir = SystemProperties.getString(WORKSPACES_DIR_PROP);
         if (newWorkspacesDir != null && !workspaceDir.equals(newWorkspacesDir)) {
-            LOGGER.log(Level.WARNING, "Changing workspaces directories from {0} to {1}. Beware that no automated data migration will occur.",
+        	Level level = workspaceDir.equals(DEFAULT_WORKSPACES_DIR) ? Level.INFO : Level.WARNING;
```

Comment:
I think this will not work. If the user changes it once, then again, the warning will not be shown when it should be.

I agree with James that writing a test for this is important here. This will make it obvious that it works or not, and will make much easier for you @charanbir to test and iterate on making sure your change does work.